apis/v1alpha2/gateway_types.go

```diff
@@ -212,17 +212,28 @@ const (
 	// Controllers may raise this condition with other reasons,
 	// but should prefer to use the reasons listed above to improve
 	// interoperability.
+	GatewayConditionAccepted GatewayConditionType = "Accepted"
+
+	// The "Scheduled" GatewayConditionType is deprecated and "Accepted" should
+	// be used instead.
+	// TODO: is there a need/way to add a kubebuilder annotation here?
 	GatewayConditionScheduled GatewayConditionType = "Scheduled"
 
+	// This reason is used with the "Accepted" condition when the condition is
+	// True.
+	GatewayReasonAccepted GatewayConditionReason = "Accepted"
+
 	// This reason is used with the "Scheduled" condition when the condition is
-	// true.
+	// True. This reason is deprecated and GatewayReasonAccepted should be used
+	// with GatewayConditionAccepted instead.
+	// TODO: is there a need/way to add a kubebuilder annotation here?
```

Comment:
I am only aware of the `kubebuilder:deprecatedversion` tag as per the [kubebuilder CRD generation documentation](https://book.kubebuilder.io/reference/markers/crd.html?highlight=deprecat#crd-generation), which does not apply here because it marks the entire version as deprecated. I am not aware of any way to call a specific field deprecated. I think we can just use the `godoc` you've already provided explaining that it's deprecated for now, maybe we can open/find an upstream issue to add something that automates this for us in kubebuilder later.

Reply:
Something similar was proposed but ended up getting turned down: https://github.com/kubernetes/enhancements/pull/3340.